### PR TITLE
Add nuget publish

### PR DESCRIPTION
**Pull request by adamskt**
Project changes to get Nuget packaging to work via the AppVeyor CI build.

NB: Because we have dependencies on pre-release Nugets (and I'm not sure we need them), we get warnings during Nuget packaging because our package versions aren't listed as pre-release as well.

To see the private AppVeyor nuget feed [read this](https://www.appveyor.com/docs/nuget/#configuring-private-nuget-feed-in-visual-studio). Our account feed: [https://ci.appveyor.com/nuget/intellitect-jamnmeof65y0](https://ci.appveyor.com/nuget/intellitect-jamnmeof65y0)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
-  - 2.0
 image: Visual Studio 2017
 configuration: Debug
-platform: Any CPU
+assembly_info:
+  patch: true
+  file: '**\AssemblyInfo.*'
+  assembly_version: '{version}'
+  assembly_file_version: '{version}'
+  assembly_informational_version: '{version}'
```
Comment (Keboo) [inline appveyor.yml]: Be aware that this does noting with the new csproj format. In this case the assembly info is generated as part of the build which occurs after app veyor attempts to find and patch the AssemblyInfo files.
I see you are handling those cases in the dotnet_csproj below. Is this still needed?

File: appveyor.yml [hunk 2]
```diff
@@ -18,9 +30,22 @@ before_build:
 
     gulp coalesce
 
-    cd ..\Coalesce.Domain
-
-    dotnet ef database update
+    cd ..\..
```
Comment (Keboo) [inline appveyor.yml]: Why this `cd`?

File: appveyor.yml [hunk 3]
```diff
@@ -18,9 +30,22 @@ before_build:
...
+  - dotnet pack "src\IntelliTect.Coalesce.CodeGeneration\IntelliTect.Coalesce.CodeGeneration.csproj" -c %CONFIGURATION% --no-build --include-symbols --output "%APPVEYOR_BUILD_FOLDER%\artifacts"
+  - dotnet pack "src\IntelliTect.Coalesce.Cli\IntelliTect.Coalesce.Cli.csproj" -c %CONFIGURATION% --no-build --include-symbols --output "%APPVEYOR_BUILD_FOLDER%\artifacts"
+artifacts:
+- path: artifacts\**\*.*
+deploy:
+  provider: NuGet
+  server: https://ci.appveyor.com/nuget/intellitect-jamnmeof65y0/api/v2/package 
+  symbol_server: https://ci.appveyor.com/nuget/intellitect-jamnmeof65y0/api/v2/package
+  api_key:
+    secure: 6BCd6W92HLY2RV4bTQSKNue9eoB7ETEYbvihtXy8T5k=
```
Comment (Keboo) [inline appveyor.yml]: Are you sure you want your nuget key in here? Typically you get around this by [creating an environment](https://www.appveyor.com/docs/deployment/environment/) and simply invoking that.

File: src/Coalesce.Web.Tests/WebUiTests.cs [hunk 4]
```diff
@@ -59,7 +59,7 @@ private void RunUiTest(IWebDriver driver)
 
             IWebElement personLink = driver.FindElement(By.PartialLinkText("Person"));
             personLink.Click();
-            driver.Manage().Timeouts().ImplicitlyWait(TimeSpan.FromSeconds(10));
+            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);
```
Comment (Keboo) [inline src/Coalesce.Web.Tests/WebUiTests.cs]: Should this be included here? Does not appear to be related to nuget stuff.
Same question for all of the code changes below too.

File: src/IntelliTect.Coalesce.CodeGeneration/IntelliTect.Coalesce.CodeGeneration.csproj [hunk 5]
```diff
@@ -1,12 +1,22 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <Description>IntelliTect.Coalesce.CodeGeneration Class Library</Description>
-    <VersionPrefix>2.0.0</VersionPrefix>
-    <Authors>intelliTect</Authors>
+    <Description>IntelliTect Coalesce code generation templates and parsers.</Description>
```
Comment (Keboo) [inline src/IntelliTect.Coalesce.CodeGeneration/IntelliTect.Coalesce.CodeGeneration.csproj]: This duplication is begging for a common targets file.

File: appveyor.yml (hunk 3 above)
Comment (adamskt) [inline appveyor.yml]: That's an encrypted version of it.  AppVeyor provides neat UI for it: https://ci.appveyor.com/tools/encrypt

File: appveyor.yml (hunk 2 above)
Comment (adamskt) [inline appveyor.yml]: because the pack commands later will fail without it :(

File: src/IntelliTect.Coalesce.CodeGeneration/IntelliTect.Coalesce.CodeGeneration.csproj (hunk 5 above)
Comment (adamskt) [inline src/IntelliTect.Coalesce.CodeGeneration/IntelliTect.Coalesce.CodeGeneration.csproj]: I agree.  I tried for a bit, and couldn't get it to work-- there is zero documentation that I could find.  Perhaps this is prime blog post fodder? :+1:
